android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageMigration.java

```diff
@@ -0,0 +1,161 @@
+package com.reactnativecommunity.asyncstorage;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.Log;
+
+import androidx.annotation.RequiresApi;
+
+import com.facebook.react.modules.storage.ReactDatabaseSupplier;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+
+// A utility class that migrates a scoped AsyncStorage database to RKStorage.
+// This utility only runs if the RKStorage file has not been created yet.
+public class AsyncStorageMigration {
+    static final String LOG_TAG = "ScopedStorageMigration";
+
+    private static Context mContext;
+
+    public static void migrate(Context context) {
+        mContext = context;
+
+        // Only migrate if the default async storage file does not exist.
+        if (isAsyncStorageDatabaseCreated()) {
+            return;
+        }
+
+        ArrayList<File> expoDatabases = getExpoDatabases();
+
+        File expoDatabase = getLastModifiedFile(expoDatabases);
```

Comment:
from the PR description:

> Get the most recently modified file to migrate. This is useful because users who change their slug and republish may have multiple database files and they'll want to use the most recent one.

to elaborate on this, as I saw it, we had two options here:

1) embed the "experienceId" ([see here](https://github.com/expo/expo/blob/9c26fbc09d2ec8c7b48d5e7d79bf99631b7b84cc/android/expoview/src/main/java/versioned/host/exp/exponent/modules/internal/ExponentAsyncStorageModule.java#L21-L24) for how it's used in managed Expo apps
2) use something like last modified time as a heuristic for determining which database was used

(2) should be sufficient and allows us to solve this problem in one place - without depending on external scripts to embed the experienceId somewhere in an ejected application, so I suggested that we proceed with this approach

Reply:
I like the second option more too